internal/cmd/keys.go

```diff
@@ -42,19 +44,22 @@ func newKeysAddCmd() *cobra.Command {
 	var options keyCreateOptions
 
 	cmd := &cobra.Command{
-		Use:   "add KEY IDENTITY",
+		Use:   "add IDENTITY",
 		Short: "Create an access key",
 		Long:  `Create an access key. Only machine identities are supported at this time.`,
 		Example: `
 # Create an access key named 'example-key' that expires in 12 hours
 $ infra keys add example-key machine-a --ttl=12h
 `,
-		Args: ExactArgs(2),
+		Args: ExactArgs(1),
 		RunE: func(cmd *cobra.Command, args []string) error {
-			keyName := args[0]
-			machineName := args[1]
+			machineName := args[0]
 
-			if strings.Contains(keyName, " ") {
+			if options.Name == "" {
+				options.Name = fmt.Sprintf("%s-%s", machineName, time.Now().Format("20060102150405.000000"))
+			}
+
+			if strings.Contains(options.Name, " ") {
```

Comment:
Should the defaulting of the name be part of the API? Would other API consumers want this same behaviour?

I suspect the validation of the name should be part of the API.

Reply:
That's a good question. I was on the fence but I think there's a good case for moving name generation to the API